modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionDemander.java

```diff
@@ -277,6 +277,9 @@ Runnable addAssignments(
         long delay = grp.config().getRebalanceDelay();
 
         if ((delay == 0 || force) && assignments != null) {
+            grp.metrics().clearRebalanceCounters();
+            grp.metrics().startRebalance(0);
```

Comment:
It is two different stages: rebalance future inited and rebalance future started.

I think we should set `rebalanceStartTime` for a cache group when the first request message sends and right after the cache group eviction happened. See the `requestPartitions` method.

WDYT?